test/prepared_test.ml

```diff
@@ -0,0 +1,70 @@
+open! Core
+open! Ctypes
+
+(* Tests for prepared statements, mimicking DuckDB's prepared statement tests *)
+
+let%expect_test "prepared_statement_basic" =
```

Comment:
There's a large number of `|> Result.ok_or_failwith`s in this test. I think all functions we're calling here where we're doing this should have an `_exn` variant which calls it for you, which will make this test more concise.